addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -98,11 +98,15 @@ function factory(dependencies) {
          */
         async remove() {
             if (!this.isTemporary) {
+                const emptyMessages = this.messages.filter(message =>
+                    !message.body || message.body.replace(/<\/?[^>]+>/g, '').trim() === ''
+                );
                 await this.async(() => this.env.services.rpc({
                     model: 'ir.attachment',
                     method: 'unlink',
                     args: [this.id],
                 }, { shadow: true }));
+                emptyMessages.map(msg => msg.delete());
```

Comment:
I don't like using `map` when the result value is irrelevant. The goal of `map` is to create a new array. Prefer normal iteration or at worse `forEach` if you want 1 line.